pgtoolkit/hba.py

```diff
@@ -147,10 +147,10 @@ def parse(cls, line: str) -> "HBARecord":
         """
         line = line.strip()
         record_fields = ["conntype", "databases", "users"]
-        values = shlex.split(line, comments=False)
+        values = [p for p in re.findall(r"(?:\".*?[^\\\\]\"|\S)+", line) if p.strip()]
```

Comment:
Can you add a comment explaining the regexp because it's totally cryptic to me (at least)?

Reply:
I added doc for the regexp.